### PR TITLE
Rename Rendering to Navigation

### PR DESCRIPTION
**Pull request by eggrobin**
Since `Frame` and `DynamicFrame` will be used for both trajectory plotting and manœuvre planning.

### CODE REVIEW

File: ksp_plugin/plugin.cpp [hunk 1]
```diff
@@ -323,13 +323,13 @@ void Plugin::UpdateFlightPlan(GUID const& vessel_guid,
...
     GUID const& vessel_guid,
-    not_null<RenderingFrame*> const rendering_frame,
+    not_null<NavigationFrame*> const navigation_frame,
     Position<World> const& sun_world_position) const {
   CHECK(!initializing_);
   not_null<std::unique_ptr<Vessel>> const& vessel =
       find_vessel_by_guid_or_die(vessel_guid);
   CHECK(vessel->is_initialized());
-  VLOG(1) << "Rendering a trajectory for the vessel with GUID " << vessel_guid;
+  VLOG(1) << "Navigation a trajectory for the vessel with GUID " << vessel_guid;
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Overzealous global substitute.

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -473,14 +473,14 @@ FrameField<World> Plugin::Navball(
           sun_world_position,
           sun_->current_position(current_time_),
           to_world.Inverse());
-  return [rendering_frame, to_world, positions_from_world, this](
+  return [navigation_frame, to_world, positions_from_world, this](
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Preference for capturing `this` first, and then the other parameters in alphabetical order.

File: ksp_plugin/plugin.hpp [hunk 3]
```diff
@@ -217,22 +217,22 @@ class Plugin {
...
-  // in |World| space as returned by |Planetarium.fetch.Sun.position|.  It is
-  // used to define the relation between |WorldSun| and |World|.
+  // |navigation_frame|.  |sun_world_position| is the current position of the
+  // sun in |World| space as returned by |Planetarium.fetch.Sun.position|.  It
+  // is used to define the relation between |WorldSun| and |World|.
   // No transfer of ownership.
   // |predicted_vessel_| must have been set, and |AdvanceTime()| must have been
   // called after |predicted_vessel_| was set.  Not const because of the stupid
-  // global variable |rendering_frame_are_operating_on_predictions_|.
+  // global variable |navigation_frame_are_operating_on_predictions_|.
```
Comment (pleroy) [inline ksp_plugin/plugin.hpp]: This stupid global variable is gone, I think, so the comment should go and the method should be made const.

File: ksp_plugin_test/interface_test.cpp [hunk 4]
```diff
@@ -475,7 +476,7 @@ TEST_F(InterfaceTest, RenderedPrediction) {
   EXPECT_CALL(*plugin_,
               RenderedPrediction(
                   kVesselGUID,
-                  check_not_null(rendering_frame),
+                  check_not_null(navigation_frame),
```
Comment (pleroy) [inline ksp_plugin_test/interface_test.cpp]: `check_not_null`?

File: serialization/journal.proto [hunk 5]
```diff
@@ -176,15 +176,15 @@ message DeletePluginSerialization {
   required Out out = 2;
 }
 
-message DeleteRenderingFrame {
+message DeleteNavigationFrame {
```
Comment (pleroy) [inline serialization/journal.proto]: Please keep the messages alphabetized (this need to move up).

Comment (eggrobin) [inline serialization/journal.proto]: Done.

File: ksp_plugin_test/interface_test.cpp (hunk 4 above)
Comment (eggrobin) [inline ksp_plugin_test/interface_test.cpp]: Now that's a name I've not heard in a long time. A long time.

File: ksp_plugin/plugin.hpp (hunk 3 above)
Comment (eggrobin) [inline ksp_plugin/plugin.hpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 2 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/plugin.cpp (hunk 1 above)
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: Done.

**Pull request merged**